notebooks/paper_analysis/city_etccdi_metrics.ipynb

```diff
@@ -901,13 +827,492 @@
+    "        '''print(get_diagnostics_filepath(diag_type, 'reanalysis', institutions, ensemble_members, variable, model, 'historical', validation_period=True))'''\n",
+    "        ref = load_zarr(get_diagnostics_filepath(diag_type, 'reanalysis', institutions, ensemble_members, variable, model, 'historical', validation_period=True)).sel(city=city)\n",
+    "        # needs units for xclim \n",
+    "        if variable == 'tasmax' or variable == 'tasmin':\n",
+    "            ref[var_name].attrs['units'] = 'K'\n",
+    "        else:\n",
+    "            ref[var_name] = ref[var_name] * 1000 # conversion to mm /day \n",
+    "            ref[var_name].attrs['units'] = 'mm day-1'\n",
+    "        # print(get_diagnostics_filepath(diag_type, 'clean', institutions, ensemble_members, variable, model, 'historical'))\n",
+    "        gcm = load_zarr(get_diagnostics_filepath(diag_type, 'clean', ds_institutions, ensemble_members, variable, model, 'historical')).sel(city=city)\n",
+    "        # print(get_diagnostics_filepath(diag_type, 'downscaled', institutions, ensemble_members, variable, model, 'historical'))\n",
+    "        downscaled = load_zarr(get_diagnostics_filepath(diag_type, 'downscaled', ds_institutions, ensemble_members, variable, model, 'historical')).sel(city=city)\n",
+    "\n",
+    "        ks_test_gcm = kstest(compute_etccdi(ref[var_name].sel(time=slice(start_year, end_year)), etccdi), \n",
+    "                            compute_etccdi(gcm[var_name].sel(time=slice(start_year, end_year)), etccdi))\n",
+    "\n",
+    "        ks_test_downscaled = kstest(compute_etccdi(ref[var_name].sel(time=slice(start_year, end_year)), etccdi), \n",
+    "                            compute_etccdi(downscaled[var_name].sel(time=slice(start_year, end_year)), etccdi))\n",
+    "        \n",
+    "        if ks_test_gcm[1] > 0.05:\n",
+    "            gcm_tests += 1\n",
+    "            gcms_indices_full[(model, etccdi)] = 1.0\n",
+    "        else: \n",
+    "            gcms_indices_full[(model, etccdi)] = 0.0\n",
+    "        if ks_test_downscaled[1] > 0.05:\n",
+    "            ds_tests += 1\n",
+    "            downscaled_indices_full[(model, etccdi)] = 1.0\n",
+    "        else: \n",
+    "            downscaled_indices_full[(model, etccdi)] = 0.0\n",
+    "    \n",
+    "    gcms_ks_full[model] = gcm_tests\n",
+    "    downscaled_ks_full[model] = ds_tests"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "20fb961b-6e04-494b-8fd3-bec382554878",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# downscaled_indices_full"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "976f54f4-317d-4bb5-bfad-3adbb1b6dd29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import ujson\n",
+    "\n",
+    "# file.write(ujson.dumps(d))\n",
+    "\n",
+    "with open('/home/jovyan/paper_data/downscaled_{city}.json'.format(city=city), 'w') as fp:\n",
```

Comment:
noting the absolute filepaths. where do these jsons come from?

Reply:
these jsons are produced when computing the ETCCDIs in `city_etccdi_metrics.ipynb`. I need to move them to the impactlab bucket